### PR TITLE
[Buttons] Add button theming category method for outlined theming

### LINKED ISSUE #5847: [Buttons] Internal issue: b/120085211
This was filed as an internal issue. If you are a Googler, please visit [b/120085211](http://b/120085211) for more details.

---
#### Internal data
- Associated internal bug: [b/120085211](http://b/120085211)

### PR DESCRIPTION
**Pull request by andrewoverton**
This PR adds the MDCButton theming category method for outlined buttons.

Closes #5847.

### CODE REVIEW

**Review by jverkoey — APPROVED** [review]

File: components/Buttons/tests/unit/ButtonsThemingTest.swift [hunk 1]
```diff
@@ -84,4 +84,52 @@ class ButtonsThemingTest: XCTestCase {
...
+    XCTAssertEqual(button.disabledAlpha,1)
+    XCTAssertEqual(button.inkColor,colorScheme.primaryColor.withAlphaComponent(0.16))
+    XCTAssertEqual(button.borderColor(for: .normal), colorScheme.onSurfaceColor.withAlphaComponent(0.12))
+    // Test shape
+    XCTAssertEqual(button.layer.cornerRadius, baselineCornerRadius, accuracy: 0.001)
+    // Test typography
+    XCTAssertEqual(button.titleFont(for: .normal), typographyScheme.button)
+    // Test remaining properties
+    XCTAssertEqual(button.minimumSize.width, 0)
+    XCTAssertEqual(button.minimumSize.height, 36)
```
Comment (codeman7) [inline components/Buttons/tests/unit/ButtonsThemingTest.swift]: Should we test the borderWidth?

File: components/Buttons/tests/unit/ButtonsThemingTest.swift [hunk 2]
```diff
@@ -84,4 +84,52 @@ class ButtonsThemingTest: XCTestCase {
...
+    XCTAssertEqual(button.titleColor(for: .disabled), colorScheme.onSurfaceColor.withAlphaComponent(0.38))
+    XCTAssertEqual(button.disabledAlpha,1)
+    XCTAssertEqual(button.inkColor,colorScheme.primaryColor.withAlphaComponent(0.16))
+    XCTAssertEqual(button.borderColor(for: .normal), colorScheme.onSurfaceColor.withAlphaComponent(0.12))
+    // Test shape
+    XCTAssertEqual(button.layer.cornerRadius, baselineCornerRadius, accuracy: 0.001)
+    // Test typography
+    XCTAssertEqual(button.titleFont(for: .normal), typographyScheme.button)
+    // Test remaining properties
+    XCTAssertEqual(button.minimumSize.width, 0)
```
Comment (romoore) [inline components/Buttons/tests/unit/ButtonsThemingTest.swift]: Drive-by comment: Please use `XCTAssertEqualWithAccuracy` for CGFloat.

File: components/Buttons/tests/unit/ButtonsThemingTest.swift (hunk 1 above)
Comment (andrewoverton) [reply components/Buttons/tests/unit/ButtonsThemingTest.swift]: I don't think so... Should we? `MDCOutlinedButtonThemer` and `MDCOutlinedButtonColorThemer` don't set `borderWidth` anywhere

Comment (romoore) [reply components/Buttons/tests/unit/ButtonsThemingTest.swift]: If the styling in the Material Guidelines shows a border, we should test for it. Unit tests are supposed to be as black-box as possible.

Comment (andrewoverton) [reply components/Buttons/tests/unit/ButtonsThemingTest.swift]: Oh wait it totally does set `borderWidth` doesn't it... I'll add it
